spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/TwitterStreamingSource.java

```diff
@@ -0,0 +1,169 @@
+  private static final String LANG = "lang";
+  private static final String TIME = "time";
+  private static final String FAVC = "favCount";
+  private static final String RTC = "rtCount";
+  private static final String SRC = "source";
+  private static final String GLAT = "geoLat";
+  private static final String GLNG = "geoLong";
+  private static final String ISRT = "isRetweet";
+
+  private static final Schema SCHEMA =
+    Schema.recordOf("tweet", Schema.Field.of(ID, Schema.of(Schema.Type.LONG)),
+                    Schema.Field.of(MSG, Schema.of(Schema.Type.STRING)),
+                    Schema.Field.of(LANG, Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+                    Schema.Field.of(TIME, Schema.nullableOf(Schema.of(Schema.Type.LONG))),
+                    Schema.Field.of(FAVC, Schema.of(Schema.Type.INT)),
+                    Schema.Field.of(RTC, Schema.of(Schema.Type.INT)),
+                    Schema.Field.of(SRC, Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+                    Schema.Field.of(GLAT, Schema.nullableOf(Schema.of(Schema.Type.DOUBLE))),
+                    Schema.Field.of(GLNG, Schema.nullableOf(Schema.of(Schema.Type.DOUBLE))),
+                    Schema.Field.of(ISRT, Schema.of(Schema.Type.BOOLEAN)));
+
+  private final TwitterStreamingConfig config;
+
+  public TwitterStreamingSource(TwitterStreamingConfig config) {
+    super(config);
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    super.configurePipeline(pipelineConfigurer);
+    pipelineConfigurer.getStageConfigurer().setOutputSchema(SCHEMA);
+  }
+
+  @Override
+  public JavaDStream<StructuredRecord> getStream(StreamingContext context) throws Exception {
+    registerUsage(context);
+    JavaStreamingContext javaStreamingContext = context.getSparkStreamingContext();
+
+    // Create authorization from user-provided properties
+    ConfigurationBuilder configurationBuilder = new ConfigurationBuilder();
+    configurationBuilder.setDebugEnabled(false)
+      .setOAuthConsumerKey(config.consumerKey)
+      .setOAuthConsumerSecret(config.consumerSecret)
+      .setOAuthAccessToken(config.accessToken)
+      .setOAuthAccessTokenSecret(config.accessTokenSecret);
+    Authorization authorization = new OAuthAuthorization(configurationBuilder.build());
+
+    return TwitterUtils.createStream(javaStreamingContext, authorization).map(
+      new Function<Status, StructuredRecord>() {
+        public StructuredRecord call(Status status) {
+          return convertTweet(status);
+        }
+      }
+    );
+  }
+
+  private StructuredRecord convertTweet(Status tweet) {
+    StructuredRecord.Builder recordBuilder = StructuredRecord.builder(SCHEMA);
+    recordBuilder.set(ID, tweet.getId());
+    recordBuilder.set(MSG, tweet.getText());
+    recordBuilder.set(LANG, tweet.getLang());
+    Date tweetDate = tweet.getCreatedAt();
+    if (tweetDate != null) {
```

Comment:
The logic is taken from the deprecated `TwitterSource` in realtime sources. The history says that @albertshau was the last to touch those lines, maybe he knows? I tried looking in the implementation of `StatusJSONImpl` but couldn't determine which fields could be nullable.